protocol/message_persistence.go

```diff
@@ -1757,21 +1758,58 @@ func (db sqlitePersistence) SavePinMessage(message *common.PinMessage) (inserted
 	return db.savePinMessage(message, queries)
 }
 
-func (db sqlitePersistence) DeleteMessage(id string) error {
-	_, err := db.db.Exec(`DELETE FROM user_messages WHERE id = ?`, id)
-	return err
+func (db sqlitePersistence) DeleteMessage(id string) (err error) {
+	tx, err := db.db.BeginTx(context.Background(), &sql.TxOptions{})
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		// don't shadow original error
+		_ = tx.Rollback()
+	}()
+
+	_, err = tx.Exec(`DELETE FROM user_messages WHERE id = ?`, id)
+	if err != nil {
+		return
+	}
+
+	_, err = tx.Exec(`DELETE FROM pin_messages WHERE message_id = ?`, id)
+
+	return
 }
 
-func (db sqlitePersistence) DeleteMessages(ids []string) error {
+func (db sqlitePersistence) DeleteMessages(ids []string) (err error) {
 	idsArgs := make([]interface{}, 0, len(ids))
 	for _, id := range ids {
 		idsArgs = append(idsArgs, id)
 	}
 	inVector := strings.Repeat("?, ", len(ids)-1) + "?"
 
-	_, err := db.db.Exec("DELETE FROM user_messages WHERE id IN ("+inVector+")", idsArgs...) // nolint: gosec
+	tx, err := db.db.BeginTx(context.Background(), &sql.TxOptions{})
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		// don't shadow original error
+		_ = tx.Rollback()
+	}()
 
-	return err
+	_, err = tx.Exec("DELETE FROM user_messages WHERE id IN ("+inVector+")", idsArgs...) // nolint: gosec
+	if err != nil {
+		return
+	}
+
+	_, err = tx.Exec("DELETE FROM pin_messages WHERE message_id IN ("+inVector+")", idsArgs...) // nolint: gosec
```

Comment:
Most intuitive would be using ON DELETE clause for that: `FOREIGN KEY (message_id) REFERENCES user_messages(id) ON DELETE CASCADE`

Migration for that would be:
```sql
CREATE TABLE pin_messages_new (
  id VARCHAR PRIMARY KEY NOT NULL,
  message_id VARCHAR NOT NULL,
  whisper_timestamp INTEGER NOT NULL,
  chat_id VARCHAR NOT NULL,
  local_chat_id VARCHAR NOT NULL,
  clock_value INT NOT NULL,
  pinned BOOLEAN NOT NULL,
  pinned_by TEXT,
  FOREIGN KEY (message_id) REFERENCES user_messages(id) ON DELETE CASCADE
);

INSERT INTO pin_messages_new (id, message_id, whisper_timestamp, chat_id, local_chat_id, clock_value, pinned, pinned_by)
SELECT id, message_id, whisper_timestamp, chat_id, local_chat_id, clock_value, pinned, pinned_by
FROM pin_messages;

DROP TABLE pin_messages;

ALTER TABLE pin_messages_new RENAME TO pin_messages;
```

Reply:
Oh I didn't know about that clause!

Is there no way to migrate a current table to use that? We need a new table absolutely?

Reply:
I asked ChatGPT and it said no :cry: 

Alright, you're migration looks clean enough! I'll apply it.

Reply:
It works, thanks!